### PR TITLE
[C-1322] Android and general performance improvements

### PR DESCRIPTION
**Pull request by sliptype**
### Description

Sorry for the big pr 😅

* use `useProxySelector` for all notifications to prevent rerenders when navigating
* Reduce rerenders of skeleton by memoizing the animation and starting the animation from the `onLayout` callback
* Increase max listeners of remote config EventEmitter to ignore warning about possible memory leak
* Reduce `DynamicImage` rerenders by memoizing styles that are passed in and only using a single onLayout callback to set the size for both images
  > This may cause issues if the sizes of the 2 images are different but I don't think we have that use case
* Reduce rerenders when playing from a lineup by pushing the play toggling logic into the lineup sagas. This allows each of the tile components to not have to know about the current playing state unless it is the currently playing track
  > This is a fairly big change and also result in us firing analytics from inside the lineup sagas. Would love thoughts here!

@dylanjeffers I reverted the navigation changes for now and will address in next pr because I was seeing the double navigation problem. Will try to use getNavigator to fix that

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/lineup-tile/CollectionTileTrackList.tsx [hunk 1]
```diff
@@ -123,7 +129,7 @@ export const CollectionTileTrackList = ({
       {tracks.slice(0, DISPLAY_TRACK_COUNT).map((track, index) => (
         <TrackItem
           key={track.uid}
-          active={playingUid === track.uid}
+          uid={track.uid}
```
Comment (dylanjeffers) [inline packages/mobile/src/components/lineup-tile/CollectionTileTrackList.tsx:132]: so the main idea here is that we don't rerender every tile when playinguid changes yeah?

File: packages/mobile/src/components/lineup-tile/LineupTileArt.tsx [hunk 2]
```diff
@@ -22,22 +23,24 @@ export const LineupTileArt = ({
   onLoad,
   style
 }: LineupTileArtProps) => {
-  const styles = useThemedStyles(createStyles)
```
Comment (dylanjeffers) [inline packages/mobile/src/components/lineup-tile/LineupTileArt.tsx:25]: nice

File: packages/mobile/src/components/lineup-tile/LineupTileCoSign.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ type Props = {
 }
 
 export const LineupTileCoSign = ({ coSign }: Props) => {
-  const trackTileStyles = useThemedStyles(createTrackTileStyles)
```
Comment (dylanjeffers) [inline packages/mobile/src/components/lineup-tile/LineupTileCoSign.tsx:57]: we should prob look to remove useThemeStyles everywhere at some point

File: packages/mobile/src/components/lineup-tile/LineupTileStats.tsx [hunk 4]
```diff
@@ -92,8 +90,8 @@ export const LineupTileStats = ({
   saveCount,
   showRankIcon
 }: Props) => {
-  const styles = useThemedStyles(createStyles)
-  const trackTileStyles = useThemedStyles(createTrackTileStyles)
+  const styles = useStyles()
+  const trackTileStyles = useTrackTileStyles()
```
Comment (dylanjeffers) [inline packages/mobile/src/components/lineup-tile/LineupTileStats.tsx:94]: this should prob help a ton

File: packages/mobile/src/components/now-playing-drawer/NowPlayingDrawer.tsx [hunk 5]
```diff
@@ -94,7 +94,7 @@ type NowPlayingDrawerProps = {
  * Memoized to prevent rerender during bottom-bar navigation.
  * It's rerendering because bottomTab render function rerenders a lot.
  */
-export const NowPlayingDrawer = memo(function NowPlayngDrawer(
+export const NowPlayingDrawer = memo(function NowPlayingDrawer(
```
Comment (dylanjeffers) [inline packages/mobile/src/components/now-playing-drawer/NowPlayingDrawer.tsx:97]: nice <3

File: packages/mobile/src/components/now-playing-drawer/TrackInfo.tsx [hunk 6]
```diff
@@ -42,12 +42,12 @@ export const TrackInfo = ({
     <View style={styles.root}>
       {user && track ? (
         <>
-          <Pressable onPress={onPressTitle}>
+          <TouchableOpacity onPress={onPressTitle}>
```
Comment (dylanjeffers) [inline packages/mobile/src/components/now-playing-drawer/TrackInfo.tsx:45]: what's the idea here? just better looking ui?

File: packages/mobile/src/components/skeleton/Skeleton.tsx [hunk 7]
```diff
@@ -20,48 +19,46 @@ type SkeletonProps = {
...
-      borderRadius: 4
-    },
-    skeleton: {
-      position: 'absolute',
-      width: '400%',
-      height: '100%',
-      backgroundColor: themeColors.skeleton
-    }
-  })
+const useStyles = makeStyles(({ palette }) => ({
```
Comment (dylanjeffers) [inline packages/mobile/src/components/skeleton/Skeleton.tsx:22]: wow... we didn't have this for skeleton, lmao rippp

File: packages/mobile/src/components/skeleton/Skeleton.tsx [hunk 8]
```diff
@@ -20,48 +19,46 @@ type SkeletonProps = {
...
   return (
     <View style={[styles.view, { height, width }, style]}>
       <Animated.View
         onLayout={(e) => {
           if (noShimmer) return
           const { width } = e.nativeEvent.layout
-          setShimmerWidth(width)
-          setShimmerPos(new Animated.Value(-0.75 * width))
+          shimmerPos.setValue(-0.75 * width)
+          shimmer.start()
```
Comment (dylanjeffers) [inline packages/mobile/src/components/skeleton/Skeleton.tsx:61]: smart so does this prevent a rerender or just speed up when we start to shimmer?

File: packages/mobile/src/screens/notifications-screen/Notifications/MilestoneNotification.tsx [hunk 9]
```diff
@@ -91,8 +95,9 @@ type MilestoneNotificationProps = {
 export const MilestoneNotification = (props: MilestoneNotificationProps) => {
   const { notification } = props
   const { achievement } = notification
-  const entity = useSelector((state) =>
-    getNotificationEntity(state, notification)
+  const entity = useProxySelector(
+    (state) => getNotificationEntity(state, notification),
+    [notification]
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/notifications-screen/Notifications/MilestoneNotification.tsx:100]: so it is the case that "notification" object is stable?

File: packages/web/src/common/store/lineup/sagas.js [hunk 10]
```diff
@@ -343,6 +345,27 @@ function* pause(action) {
   yield put(queueActions.pause({}))
 }
 
+function* togglePlay(lineupActions, lineupSelector, prefix, action) {
```
Comment (dylanjeffers) [inline packages/web/src/common/store/lineup/sagas.js:347]: this makes sense to me

**Review by dylanjeffers — APPROVED** [review]
awesome work, stoked to see the improvements in dynamic image and skeleton in particular. big lol that we were `Stylesheet.create`ing 100s of times every page. Notif work is great too

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-android-perf

File: packages/mobile/src/components/lineup-tile/CollectionTileTrackList.tsx (hunk 1 above)
Comment (sliptype) [reply packages/mobile/src/components/lineup-tile/CollectionTileTrackList.tsx:132]: Yes, this line is specific to the items in the collection tile but the other changes in the pr achieve what you said

File: packages/mobile/src/components/lineup-tile/LineupTileCoSign.tsx (hunk 3 above)
Comment (sliptype) [reply packages/mobile/src/components/lineup-tile/LineupTileCoSign.tsx:57]: I'm down, there are a lot of uses though

File: packages/mobile/src/components/now-playing-drawer/TrackInfo.tsx (hunk 6 above)
Comment (sliptype) [reply packages/mobile/src/components/now-playing-drawer/TrackInfo.tsx:45]: Yeah it felt weird to have the pressable things with no feedback

File: packages/mobile/src/components/lineup-tile/LineupTileCoSign.tsx (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/lineup-tile/LineupTileCoSign.tsx:57]: yeah the whole sign-up flow could use a perf facelift 🗡️

File: packages/mobile/src/components/skeleton/Skeleton.tsx (hunk 8 above)
Comment (sliptype) [reply packages/mobile/src/components/skeleton/Skeleton.tsx:61]: Prevents a render bc we aren't setting width in state

File: packages/mobile/src/screens/notifications-screen/Notifications/MilestoneNotification.tsx (hunk 9 above)
Comment (sliptype) [reply packages/mobile/src/screens/notifications-screen/Notifications/MilestoneNotification.tsx:100]: I believe so, this is how the other notif types are doing it

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-android-perf

File: packages/web/src/common/store/lineup/sagas.js (hunk 10 above)
Comment (nicoback2) [reply packages/web/src/common/store/lineup/sagas.js:347]: @sliptype lol rip i was doing the exact same thing in my branch 💀. but the good news is we both thought this was a good idea

Comment (sliptype) [reply packages/web/src/common/store/lineup/sagas.js:347]: Oh noooo, sorry I should have communicated that

**Pull request merged**